src/snowflake/snowpark/modin/pandas/general.py

```diff
@@ -394,26 +394,158 @@ def merge_ordered(
+        a left_val  right_val
+    0   1        a        NaN
+    1   5        b        3.0
+    2  10        c        7.0
+    >>> pd.merge_asof(left, right, on="a", direction="forward")
+        a left_val  right_val
+    0   1        a        1.0
+    1   5        b        6.0
+    2  10        c        NaN
+
+    Here is a real-world times-series example:
+
+    >>> quotes = pd.DataFrame(
+    ...    {
+    ...        "time": [
+    ...            pd.Timestamp("2016-05-25 13:30:00.023"),
+    ...            pd.Timestamp("2016-05-25 13:30:00.023"),
+    ...            pd.Timestamp("2016-05-25 13:30:00.030"),
+    ...            pd.Timestamp("2016-05-25 13:30:00.041"),
+    ...            pd.Timestamp("2016-05-25 13:30:00.048"),
+    ...            pd.Timestamp("2016-05-25 13:30:00.049"),
+    ...            pd.Timestamp("2016-05-25 13:30:00.072"),
+    ...            pd.Timestamp("2016-05-25 13:30:00.075")
+    ...        ],
+    ...        "bid": [720.50, 51.95, 51.97, 51.99, 720.50, 97.99, 720.50, 52.01],
+    ...        "ask": [720.93, 51.96, 51.98, 52.00, 720.93, 98.01, 720.88, 52.03]
+    ...    }
+    ... )
+    >>> quotes
+                         time     bid     ask
+    0 2016-05-25 13:30:00.023  720.50  720.93
+    1 2016-05-25 13:30:00.023   51.95   51.96
+    2 2016-05-25 13:30:00.030   51.97   51.98
+    3 2016-05-25 13:30:00.041   51.99   52.00
+    4 2016-05-25 13:30:00.048  720.50  720.93
+    5 2016-05-25 13:30:00.049   97.99   98.01
+    6 2016-05-25 13:30:00.072  720.50  720.88
+    7 2016-05-25 13:30:00.075   52.01   52.03
+    >>> trades = pd.DataFrame(
+    ...    {
+    ...        "time": [
+    ...            pd.Timestamp("2016-05-25 13:30:00.023"),
+    ...            pd.Timestamp("2016-05-25 13:30:00.038"),
+    ...            pd.Timestamp("2016-05-25 13:30:00.048"),
+    ...            pd.Timestamp("2016-05-25 13:30:00.048"),
+    ...            pd.Timestamp("2016-05-25 13:30:00.048")
+    ...        ],
+    ...        "price": [51.95, 51.95, 720.77, 720.92, 98.0],
+    ...        "quantity": [75, 155, 100, 100, 100]
+    ...    }
+    ... )
+    >>> trades
+                         time   price  quantity
+    0 2016-05-25 13:30:00.023   51.95        75
+    1 2016-05-25 13:30:00.038   51.95       155
+    2 2016-05-25 13:30:00.048  720.77       100
+    3 2016-05-25 13:30:00.048  720.92       100
+    4 2016-05-25 13:30:00.048   98.00       100
+    >>> pd.merge_asof(trades, quotes, on="time")
+                         time   price  quantity     bid     ask
+    0 2016-05-25 13:30:00.023   51.95        75   51.95   51.96
+    1 2016-05-25 13:30:00.038   51.95       155   51.97   51.98
+    2 2016-05-25 13:30:00.048  720.77       100  720.50  720.93
+    3 2016-05-25 13:30:00.048  720.92       100  720.50  720.93
+    4 2016-05-25 13:30:00.048   98.00       100  720.50  720.93
     """
     # TODO: SNOW-1063345: Modin upgrade - modin.pandas functions in general.py
     if not isinstance(left, DataFrame):
```

Comment:
I think that this check may be wrong? We're checking that the left isn't of type `DataFrame`,  and then erroring out on the type of the right? We should probably be checking the type of the right as well?

Reply:
No I think this is correct. We are checking that the left value is a DataFrame, if it isn't we are saying that we can't merge it with the right value's type.

Reply:
but shouldn't we be saying that since it isn't a DataFrame, we can't merge type(left)? (Purely semantic, but this is how the other error messages usually are, but maybe this is a weird pandas thing)?

Reply:
Fair point, changed it to `left` and added a similar check for `right` like you suggested.